test/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializerTest.java

```diff
@@ -642,4 +653,35 @@ public int hashCode () {
 			return Objects.hash(value, list, serializable);
 		}
 	}
+
+	@Test
+	public void testLogWarningOnDuplicateFieldInClassHierarchy() {
+		kryo.setReferences(true);
+		CompatibleFieldSerializer serializer = new CompatibleFieldSerializer(kryo, ClassWithDuplicateField.class);
+		serializer.getCompatibleFieldSerializerConfig().setChunkedEncoding(true);
+		serializer.getCompatibleFieldSerializerConfig().setExtendedFieldNames(false);
+		serializer.updateFields();
+		kryo.register(ClassWithDuplicateField.class, serializer);
+
+		final ClassWithDuplicateField duplicateField = new ClassWithDuplicateField();
+		roundTrip(31, duplicateField);
+
+		ClassWithDuplicateField object2 = (ClassWithDuplicateField) kryo.readClassAndObject(input);
+		assertEquals(duplicateField, object2);
```

Comment:
This `equals` does not make much sense because it does not compare on the `customNote` field. You should include the field in the `equals` contract. What happens then? Does the test fail? 

As I mentioned before, we should also assert that a warning is actually logged. Please see `WarnUnregisteredClassesTest` for how this can be done.

Reply:
Agree！Done.